### ISSUE TITLE
When `referenceElement` prop is set to `undefined`, `referenceProps.ref` no longer works.

### ISSUE THREAD

**Issue by HHogg** [FezVrasta/react-popper#248]
### Reproduction demo

 https://codesandbox.io/s/xpyk0364op

### What went wrong?

When `referenceElement` prop is set to `undefined`, `referenceProps.ref` no longer works.

There are some useful situations to conditionally use `referenceElement`.

### What is the expected behavior?

I think this is probably expected behaviour at the moment, but the doc confused me a little bit with 

> If referenceElement is defined, it will take precedence over any referenceProps.ref provided refs.

I thought with `referenceElement` set to `undefined` then `referenceProps.ref` would work as expected. 

### Any other comments?

It's not a massive issue as you can get around it by doing.

```jsx
const { referenceElement } = this.props;
const popperProps = {};

if (referenceElement) {
  popperProps.referenceElement = referenceElement;
}

return ( 
  <Popper { ...popperProps }>...</Popper>
);
```

.... just a little nasty that's all. 

### Packages versions

- Popper.js: 1.14.4
- react-popper: 1.0.2

**Comment by gwyneplaine** [comment]
@HHogg this should be fixed in the latest version of react-popper, 
as a fix for this was merged in to master a few weeks ago. 

https://github.com/FezVrasta/react-popper/pull/231

Though i'm not sure which release this was published under, @FezVrasta might be of more help there.

**Comment by FezVrasta** [comment]
Released as 1.3.1
